flex-checks/src/main/java/org/sonar/flex/checks/ClassComplexityCheck.java

```diff
@@ -63,11 +62,8 @@ public void leaveNode(AstNode node) {
     SourceClass sourceClass = (SourceClass) getContext().peekSourceCode();
     int complexity = ChecksHelper.getRecursiveMeasureInt(sourceClass, FlexMetric.COMPLEXITY);
     if (complexity > maximumClassComplexityThreshold) {
-      getContext().createLineViolation(this,
-        "Class has a complexity of {0,number,integer} which is greater than {1,number,integer} authorized.",
-        node,
-        complexity,
-        maximumClassComplexityThreshold);
+      String message = String.format("Class has a complexity of %s which is greater than %s authorized.", complexity, maximumClassComplexityThreshold);
+      createIssueWithCost(message, node, complexity - maximumClassComplexityThreshold);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this substraction operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2184)